lib/datadog/appsec/waf/lib_ddwaf.rb

```diff
@@ -95,7 +95,11 @@ def self.libddwaf_dir
         end
 
         def self.shared_lib_extname
-          Gem::Platform.local.os == 'darwin' ? '.dylib' : '.so'
+          if Gem::Platform.local.os == 'darwin' || Gem::Platform.local.os == 'java' && java.lang.System.get_property('os.name').match(/mac/i)
```

Comment:
Honestly, I would go with `elsif` here, it's too long and too cumbersome to understand

Reply:
Do you think that the second condition after the `||` could just be on a new line instead of a separate elsif ? I  don't feel like having two different `if` with the same result would be less cumbersome to understand than just a newline (but I agree that both would be better than currently)

Reply:
The problem here is purely human related, it's so long to read that I forget what I was reading in the middle of the line.

Less mental load, smaller code, easier to understand. Our goal here is to have understandable code, since we don't care about performance

Reply:
Fixed in 40c8db31da61bef1f23562982f5bd629e60962b1